tools/cleanup_duplicate_host_guest_devices.rb

```diff
@@ -45,6 +46,22 @@
 
 return if opts[:dry_run]
 
-GuestDevice.destroy(guest_devices_to_delete) unless opts[:dry_run]
+associations = %i[has_one has_many].flat_map { |assoc| GuestDevice.reflect_on_all_associations(assoc) }
+dependents   = associations.select { |assoc| assoc.options[:dependent].present? }
```

Comment:
out of scope but...

do we turn this into a function? (a recursive function?) :)